llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -745,6 +747,37 @@ class OpLowerer {
     });
   }
 
+  [[nodiscard]] bool lowerLifetimeIntrinsic(Function &F) {
+    IRBuilder<> &IRB = OpBuilder.getIRB();
+    return replaceFunction(F, [&](CallInst *CI) -> Error {
```

Comment:
why do we need a lambda for this?

Reply:
I could hoist the `[&](CallInst *CI) -> Error {...}` function out to a static module level function and then pass it in instead...
Or avoid using `replaceFunction` and iterate over all users and manually replace the users, but then it's basically duplicating the code of `replaceFunction` itself.

Reply:
Is it ok to keep this as is? Or should I replace it with one of the two alternatives I mentioned?

Reply:
Leave it as is. I didn’t like the alternatives.